### ISSUE TITLE
azurerm_storage_queue stored as table storage in State

### ISSUE THREAD

**Issue by rdvansloten** [hashicorp/terraform-provider-azurerm#25298]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.4.7

### AzureRM Provider Version

3.96.0

### Affected Resource(s)/Data Source(s)

azurerm_storage_queue

### Terraform Configuration Files

```hcl
resource "azurerm_storage_account" "main" {
  name                     = "mystorageaccount"
  resource_group_name      = azurerm_resource_group.main.name
  location                 = azurerm_resource_group.main.location
  account_tier             = "Standard"
  account_replication_type = "LRS"
  account_kind             = "StorageV2"
}

resource "azurerm_storage_queue" "message" {
  name                 = "myqueue"
  storage_account_name = azurerm_storage_account.main.name
}
```

### Debug Output/Panic Output

```shell
azurerm_storage_queue.message: Creating...
╷
│ Error: expected the subdomain type to be "queue" but got "table"
│ 
│   with azurerm_storage_queue.message,
│   on main.tf line 63, in resource "azurerm_storage_queue" "message":
│   63: resource "azurerm_storage_queue" "message" {
```

### Expected Behaviour

It should create a Queue and it does, but not in a good way.

### Actual Behaviour

It's creating a Queue (visible in Portal and state), but it crashes the subsequent run because the URL in the state does not match:

```
{
      "mode": "managed",
      "type": "azurerm_storage_queue",
      "name": "message",
      "provider": "provider[\"registry.terraform.io/hashicorp/azurerm\"]",
      "instances": [
        {
          "status": "tainted",
          "schema_version": 1,
          "attributes": {
            "id": "https://mystorageaccount.table.core.windows.net/myqueue",
            "metadata": null,
            "name": "myqueue",
            "resource_manager_id": null,
            "storage_account_name": "mystorageaccount",
            "timeouts": null
          },
          "sensitive_attributes": [],
          "private": "REDACTED",
          "dependencies": [
            "azurerm_storage_account.main"
          ]
        }
      ]
    }
```

### Steps to Reproduce

Apply the configuration given above with the versions listed.

### Important Factoids

No

**Comment by rdvansloten** [comment]
Seems to work fine if I specify < 3.96 for the azurerm version.

**Comment by rcskosir** [comment]
@rdvansloten Thank you for taking the time to open this issue. If I am not mistaken, I believe this is a duplicate of #25261 which is solved via #25262 which has been merged and will go out in the next release. As such, I'm going to mark this as a duplicate and close it with #25262. If I am mistaken, please let me know and I can reopen.
